File: data/src/main/java/ch/srg/dataProvider/integrationlayer/data/IlImage.java
package ch.srg.dataProvider.integrationlayer.data;

import androidx.annotation.NonNull;

import java.util.Objects;

/**
 * Copyright (c) SRG SSR. All rights reserved.
 * <p>
 * License information is available from the LICENSE file.
 */
public final class IlImage {

    public enum Size {
        w240(240),
        w320(320),
        w480(480),
        w960(960),
        w1920(1920);
        public final int value;

        Size(int size) {
            this.value = size;
        }

        public static Size getClosest(int pixels) {
            if (pixels >= w1920.value) {
                return w1920;
            }
            if (pixels <= w240.value) {
                return w240;
            }
            Size[] sizes = Size.values();
            int closestSize = 0;
            int minDist = Integer.MAX_VALUE;
            for (int i = 0; i < sizes.length; i++) {
                int dist = Math.abs(sizes[i].value - pixels);
                if (dist <= minDist) {
                    minDist = dist;
                    closestSize = i;
                }
            }
            return sizes[closestSize];
        }
    }

    @NonNull
    private final String url;

    public IlImage(@NonNull String url) {
        this.url = url;
    }

    @NonNull
    public String getUrl() {
        return url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        IlImage ilImage = (IlImage) o;
        return url.equals(ilImage.url);
    }

    @Override
    public int hashCode() {
        return Objects.hash(url);
    }
}

File: data/src/main/java/ch/srg/dataProvider/integrationlayer/data/IlImage.kt
package ch.srg.dataProvider.integrationlayer.data

/**
 * Copyright (c) SRG SSR. All rights reserved.
 *
 *
 * License information is available from the LICENSE file.
 */
data class IlImage @JvmOverloads constructor(val url: String) {

    @Suppress("MagicNumber")
    enum class Size(val sizePixels: Int) {
        W240(240), W320(320), W480(480), W960(960), W1920(1920);

        companion object {
            fun getClosest(pixels: Int): Size {
                if (pixels >= W1920.sizePixels) {
                    return W1920
                }
                if (pixels <= W240.sizePixels) {
                    return W240
                }
                val sizes = values()
                var closestSize = 0
                var minDist = Int.MAX_VALUE
                for (i in sizes.indices) {
                    val dist = Math.abs(sizes[i].sizePixels - pixels)
                    if (dist <= minDist) {
                        minDist = dist
                        closestSize = i
                    }
                }
                return sizes[closestSize]
            }
        }
    }
}
